### ISSUE TITLE
Add Support for GraphQL Yoga

### ISSUE THREAD

**Issue by varshith257** [PalisadoesFoundation/talawa-api#2028]
**Is your feature request related to a problem? Please describe.**

The current Apollo Server lacks stability on a single version around it going on release versions and every previous version will go deprecated once a new version is released. It needs to update Apollo Server in our project for every version it releases.

**Describe the solution you'd like**

1. Instead Apollo Server add support for GraphQL Yoga in our project.  It's built on web standards, supports all modern features, less bloated compared to apollo-server, keeps itself minimal and lets the user choose what they want to use.
2. It's also officially maintained by [the-guild](https://the-guild.dev/#platform). They're the ones responsible for maintaining most of the well-known packages in the javascript graphql ecosystem.
3. With graphql-yoga you can make use of a plugin architecture to make use of the plethora of [envelop graphql plugins](https://the-guild.dev/graphql/envelop/plugins) to integrate different functionalities in an easy way.

**Describe alternatives you've considered**

- N/A

**Approach to be followed (optional)**

Remove current Apollo Server support in the project and add full support of using GraphQL Yoga

**Additional context**

It is a part of : 
  - #1997 

 **Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

**Comment by varshith257** [comment]
@palisadoes   The Fastify migration issue encountered a roadblock with Apollo Server(v4) not [supporting Fastify](https://www.apollographql.com/docs/apollo-server/migration#removed-integrations) 

1. If I have to work on GraphQL Yoga Migration it has to be supported with fastify and fastify support work is currently WIP.
2. If I have to work on fastify to support as backend framework in project, the Apollo Server v4 removed integration of fastify.  

 It's indeed a tricky situation where Apollo Server's lack of Fastify support is blocking progress on both fronts. 

So the workaround solution to get rid of this dependency loop is to work on this two issues parallely. Already I moved towards this approach and working on it.

**Comment by AVtheking** [comment]
@varshith257 if you need help in migration to graphql-yoga then I can help in the migration and you can  continue on fastify migration.

**Comment by varshith257** [comment]
@AVtheking Thanks, but working on both parallely for above reasons.

**Comment by AVtheking** [comment]
okay 👍

**Comment by varshith257** [comment]
@palisadoes Is this to work with the present express framework or this also need a hold?

**Comment by palisadoes** [comment]
Closing as we need to put this on hold again

**Comment by xoldd** [comment]
> To tackle the GraphQL Yoga Migration, it needs to be compatible with Fastify, but Fastify support is still a work in progress.
Graphql yoga works with expressjs as well, it is not dependent on fastify.

**Comment by varshith257** [comment]
@xoldd Yes, but before we have decided that instead going of migrating with express and then to fastify, thought to tackle both migrations parallely. 

@palisadoes decided to hold for now and after GSOC selected projects are announced, collaboratively with selected mentees as group effort can be tackle easily instead going migrations now and introducing chaos before GSOC as mentioned by Peter.

**Comment by palisadoes** [comment]
Correct.

We need to prioritize getting the remaining Admin and Mobile screens redesigned and / or functioning as expected with the appropriate tests.

Then we can start work on correcting the deprecated and dormant packages, preferably with input from the selected GSoC people because of its effect on their projects. Hopefully we’ll then be able to get others to work in parallel with supporting work.
